content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Connection management
+metaTitle: Connection management (Guide)
+metaDescription: 'How to approach connection management with Prisma Client in serverless and PaaS applications.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Databases can handle a limited number of concurrent connections. Each connection requires RAM, which means that simply increasing the connection limit without scaling the database:
+
+* ✔ might allow more processes to connect _but_
+* ✘ significantly affects database **performance**
+
+The way your application **manages connections** also impacts performance. This guide describes how to approach connection management in [serverless environments](#serverless-environments-faas) and [long-running processes](#long-running-processes-paas).
+
+</TopBlock>
+
+## Long-running processes (PaaS)
+
+Examples of long-running processes include Node.js applications hosted on a PaaS service like Heroku. Use the following checklist as a guide to connection management in PaaS environments:
+
+- ☐ Tune the [connection limit](#recommended-connection-limit)
+- ☐ Make sure you have [**one** global instance of `PrismaClient`](#prismaclient-in-long-running-applications)
+
+### Recommended connection limit
+
+The recommended connection limit for long-running processes is the **default connection limit** (`num_physical_cpus * 2 + 1`) ÷ **number of application instances**.
```

Comment:
The default connection limit or default connection limit, where does this comes from as the "recommended limit"? It is quite a random value that we default to in the Engines.

Reply:
Changed to 'connection pool size'.

As far as I know, cpus * 2 + 1 is based on `((core_count * 2) + effective_spindle_count)`, which is just a generally accepted formula. I'll change the wording to 'recommended starting point' to indicate that it's not a catch-all

Reply:
Ah. I'll make a not for someone to investigate that a bit, because: https://dba.stackexchange.com/a/228666